### PR TITLE
Handle job response

### PR DESCRIPTION
**Pull request by BradHudson**
We were receiving the following error: ***TypeError: no implicit conversion of String into Integer***

Here is the json response we were getting on the ***error*** page:

````
{"Results"=>
  {"JobRecommendation"=>
    {"Jobs"=>
      {"CompanyJob"=>
        {"JobDID"=>"JHR5RK7870G6J01G6CZ", "JobLink"=>"/JobSeeker/Jobs/JobDetails.aspx?job_did=JHR5RK7870G6J01G6CZ", "JobTitle"=>"IT Technician", "Industries"=>{"string"=>"Retail"}, "Location"=>"US-GA-Atlanta", "Source"=>nil, "Date"=>"2015-10-29T08:17:27.423"}}}}}
````

We take the ````CompanyJob```` node and ````.each```` through it. Notice that this is NOT AN ARRAY so when we would loop through it, we would not pass the entire Job to  ````Models::Job````. Pages that work correctly return an array of jobs.

***Solution***: We are going to see if it is an array, and if it not, turn it into an array.

***Testing***: There is already a test that assigns a hash to the value of CompanyJob, so we are good.

### CODE REVIEW

File: lib/cb/clients/recommendation.rb [hunk 1]
```diff
@@ -69,7 +69,9 @@ def self.for_company(company_did)
         if json_hash
           api_jobs = json_hash.fetch('Results', {}).fetch('JobRecommendation', {}).fetch('Jobs', {})
           if api_jobs
-            api_jobs.fetch('CompanyJob', []).each do |cur_job|
+            company_jobs = api_jobs.fetch('CompanyJob', [])
+            company_jobs = [company_jobs] unless company_jobs.is_a?(Array)
+            company_jobs.each do |cur_job|
```
Comment (ausmith) [inline lib/cb/clients/recommendation.rb]: So, there's another edge case that this doesn't handle: what if we fetch `'CompanyJob'` and it returns `nil` like `'Jobs'` did? There is some nifty ruby that you could do instead though!

```ruby
company_jobs = [api_jobs.fetch('CompanyJob', [])].flatten.compact
company_jobs.each ...
```

The `flatten` will ensure your array is only 1 level deep. The `compact` will remove any `nil`s.

Comment (usbsnowcrash) [inline lib/cb/clients/recommendation.rb]: Do we need a test for this edge case?

Comment (ausmith) [inline lib/cb/clients/recommendation.rb]: :+1: good idea IMO

Comment (BradHudson) [inline lib/cb/clients/recommendation.rb]: That is some nifty ruby...so wise...let me implement this and good eye

Comment (BradHudson) [inline lib/cb/clients/recommendation.rb]: I will add a test for this.

**Comment by oochelz** [comment]
Please add a test! You can use your example in the test, even.

**Comment by oochelz** [comment]
Of course when I post that, GitHub shows me comments about testing...

File: lib/cb/clients/recommendation.rb (hunk 1 above)
Comment (Casao) [inline lib/cb/clients/recommendation.rb]: You can also just use `Array(api_jobs.fetch)` which will wrap it if it's not an array, and leave it alone if it is.

Comment (ausmith) [inline lib/cb/clients/recommendation.rb]: Oh, then you wouldn't need to do the `flatten`. :+1:

Comment (ausmith) [inline lib/cb/clients/recommendation.rb]: Actually, `Array` won't do exactly what we want:

```ruby
irb(main):009:0> Array({a: 1, b: 2})
=> [[:a, 1], [:b, 2]]
```

Comment (Casao) [inline lib/cb/clients/recommendation.rb]: Ah, didn't realize it was coming back as a hash, my bad.

**Comment by BradHudson** [comment]
So I changed the original tests to already have the array.
- I then made a test for when it gets a hash
- I made a test for the edge case of CompanyJobs => nil

File: spec/cb/clients/recommendation_spec.rb [hunk 2]
```diff
@@ -102,6 +102,30 @@ module Cb
...
+    context '.for_company when CompanyJob is nil' do
+      before do
+        stub_request(:get, uri_stem(Cb.configuration.uri_recommendation_for_company))
+            .to_return(body: { Results: { JobRecommendation: { Jobs: { CompanyJob: nil } } } }.to_json)
+      end
+
+      it 'should get not have any recommendations' do
+        recs = Cb.recommendation.for_company('fake-did')
+
+        expect(recs.count).to be 0
```
Comment (ausmith) [inline spec/cb/clients/recommendation_spec.rb]: `be` or `eq`?

**Comment by tkling** [comment]
@BradHudson just a heads up - this is a public repo and links to a private New Relic account might not be appropriate here.

**Comment by BradHudson** [comment]
@tkling good call! removed.

File: spec/cb/clients/recommendation_spec.rb (hunk 2 above)
Comment (BradHudson) [inline spec/cb/clients/recommendation_spec.rb]: i think eq is more appropriate. good catch.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@ Version History
     * All Version bumps are required to update this file as well!!
 ----
 
+* 18.5.5 Turn company recs response into an array if its not.
```
Comment (ausmith) [inline CHANGELOG.md]: "it's" :stuck_out_tongue:
